src/rpcclient/rpcclient/fs.py

```diff
@@ -360,13 +370,28 @@ def read_file(self, file: str) -> bytes:
 
     @path_to_str('remote')
     @path_to_str('local')
-    def _pull_file(self, remote: str, local: str):
+    def _pull_file(self, remote: str, local: str, with_progress: bool):
         with open(local, 'wb') as local_file:
```

Comment:
the context-managers here are starting to get really messy. let's do something like that:
https://stackoverflow.com/questions/3024925/create-a-with-block-on-several-context-managers

Reply:
fixed